.github/repochanges/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"fmt"
+	"github.com/urfave/cli"
+	"gopkg.in/yaml.v3"
+	"os"
+	"sort"
+	"text/template"
+)
+
+func main() {
+
+	app := &cli.App{
+		Name:      "repochanges",
+		Version:   "0.0.1",
+		Authors:   []cli.Author{{Name: "Kairos authors"}},
+		Usage:     "Extract changes betweeen 2 luet repository version list files",
+		UsageText: `repochanges OLDREFENRENCE NEWREFERENCE`,
+		Copyright: "kairos authors",
+		ArgsUsage: "oldreference newreference",
+		Before: func(context *cli.Context) error {
+			if context.NArg() != 2 {
+				return fmt.Errorf("not enough arguments")
+			}
+			return nil
+		},
+		Action: func(c *cli.Context) error {
+			var changes []Changed
+
+			file1, err := os.ReadFile(c.Args().Get(0))
+			if err != nil {
+				return err
+			}
+			file2, err := os.ReadFile(c.Args().Get(1))
+			if err != nil {
+				return err
+			}
+
+			var oldVersions []Package
+			var newVersions []Package
+
+			_ = yaml.Unmarshal(file1, &oldVersions)
+			_ = yaml.Unmarshal(file2, &newVersions)
+
+			for _, p := range newVersions {
+				in, oldpkg := isIn(p, oldVersions)
+				if !in {
+					// New package
+					changes = append(changes, Changed{
+						Name:     p.Name,
+						Category: p.Category,
+						From:     "0",
+						To:       p.Version,
+						New:      true,
+					})
+				} else {
+					// Updated
+					if p.Version != oldpkg.Version {
+						changes = append(changes, Changed{
+							Name:     p.Name,
+							Category: p.Category,
+							From:     oldpkg.Version,
+							To:       p.Version,
+							New:      false,
+						})
+					}
+				}
+			}
+
+			// Sort by newer package first
```

Comment:
Im sure we could do this on pure bash, but the pure bash + earthly on top was killing me